utils/io/write_test.go

```diff
@@ -0,0 +1,34 @@
+package io
+
+import (
+	"errors"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTerminateOnFullDisk(t *testing.T) {
+	// happy path
+	t.Run("return nil on no error", func(t *testing.T) {
+		result := TerminateOnFullDisk(nil)
+		require.NoError(t, result)
+	})
+	t.Run("benign non disk related error should return the error", func(t *testing.T) {
+		benignError := errors.New("benign error")
+		result := TerminateOnFullDisk(benignError)
+		require.ErrorIs(t, result, benignError)
+	})
+	// sad path
+	t.Run("panic on full disk", func(t *testing.T) {
+		// imitate badgerDB error wrapping
+		badgerDiskFullError := syscall.ENOSPC
```

Comment:
I'm assuming Badger wraps the error - in which case it's better to mimic this in the test case. Though, we should confirm exactly what error Badger is throwing in this case, like Peter mentioned. 
```suggestion
		badgerDiskFullError := fmt.Errorf("this is intended to mimic an internal Badger error wrapping an out-of-disk system error: %w", syscall.ENOSPC)
```